src/controllers/tripRequestController.js

```diff
@@ -0,0 +1,173 @@
+            const trip = await tripRequests.findOne({ where: { id: requestId } })
+            if (!trip) {
+                return res.status(404).json({ status: 'fail', message: `there are no Trip Request found` })
+            } else {
+                return res.status(200).json({ status: 'success', data: trip })
+            }
+        } else {
+            return res.status(403).json({ status: 'fail', message: 'UnAuthorized' })
+        }
+    } catch (error) {
+        res.status(500).json({ status: 'error', message: 'error while getting trip request' })
+    }
+};
+
+export const getAllTripRequest = async (req, res) => {
+    try {
+        const userId = req.user.id;
+
+        if (req.user.role == 'requester') {
+            const trips = await tripRequests.findAll({ where: { requesterId: userId } })
+
+            if (!trips) {
+                return res.status(404).json({ status: 'fail', message: `there are no Trip Requests assigned to ${userId}` })
+            } else {
+                return res.status(200).json({ status: 'success', data: trips })
+            }
+
+        } else if (req.user.role == 'manager') {
+            const trips = await tripRequests.findAll()
+
+            if (!trips) {
+                return res.status(404).json({ status: 'fail', message: `there are no Trip Requests found` })
+            } else {
+                return res.status(200).json({ status: 'success', data: trips })
+            }
+        } else {
+            return res.status(403).json({ status: 'fail', message: 'you are not allowed to retrieve a trip request' })
+        }
+    } catch (error) {
+        res.status(500).json({ status: 'error', message: 'error while getting trip request' })
+    }
+}
+
+export const updateTripRequest = async (req, res) => {
+    try {
+        if (req.user.role !== 'requester') {
+            return res.status(403).json({ status: 'fail', message: 'you are not allowed to update a trip request' })
+        }
+        const requestId = req.params.id;
+        const userId = req.user.id;
+
+        const tripRequest = await tripRequests.findOne({
+            where: { id: requestId }
+        })
+
+
+        if (!tripRequest) {
+            return res.status(404).json({ status: 'fail', message: `No Trip Request with id = ${requestId}` })
+        }
+
+        if (tripRequest.status == 'pending') {
+            await tripRequestSchema.validateAsync(req.body);
+            const type = req.body.returnDate == null ? 'one way trip' : 'return type trip';
+            const returnDate = type == 'one way trip' ? null : req.body.returnDate;
+
+            const updatedTrip = {
+                leavingFrom: req.body.leavingFrom,
+                goingTo: req.body.goingTo,
+                travelDate: req.body.travelDate,
+                returnDate: returnDate,
+                travelReason: req.body.travelReason,
+                tripType: type,
+                status: req.body.stastus,
```

Comment:
You have a typo here
shouldn't this be req.body.status

Reply:
type and status have to be different
eg: type:one-way-trip,return-type-trip 
      status:pending,rejected,approved

Reply:
I meant on line 124
You have; `status: req.body.stastus` instead of `status: req.body.status`